components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authcontext/JWTTokenGenerator.java

```diff
@@ -226,6 +227,10 @@ public void generateToken(OAuth2TokenValidationMessageContext messageContext) th
         claimsSetBuilder.claim(API_GATEWAY_ID + "/enduser", authzUser);
         //TODO: check setting audience
 
+        if (accessTokenDO.getTokenType() != null) {
```

Comment:
Rather than adding it in this JWT, wouldn't it be better if we put this value in the introspection response?

Is this JWT generator used in APIM context?

Reply:
JWT generator is not used in the APIM context. We don't use this claim from the introspection response, added it to JWT. +1 to add it as a standard claim to introspection response. fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1459/commits/66a23338204aacc1be2b5a3969fa2bd518322e55